datafusion/sql/src/statement.rs

```diff
@@ -95,11 +95,22 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
     /// Generate a logical plan from an SQL statement
     pub fn sql_statement_to_plan(&self, statement: Statement) -> Result<LogicalPlan> {
-        self.sql_statement_to_plan_with_context(statement, &mut PlannerContext::new())
+        self.sql_statement_to_plan_with_context_impl(
+            statement,
+            &mut PlannerContext::new(),
+        )
     }
 
     /// Generate a logical plan from an SQL statement
-    fn sql_statement_to_plan_with_context(
+    pub fn sql_statement_to_plan_with_context(
```

Comment:
Could you rename `sql_statement_to_plan_with_context_impl` to `sql_statement_to_plan_with_context` and make it `pub` and get the same effect with less code?

Reply:
maybe the author didn't want to change the api? and only want to make it public?

Reply:
👍  I think this PR could be merged in as is. @waynexia  feel free to merge it (or let us know you plan to make more changes)